frontend/controller/router.js

```diff
@@ -194,8 +194,9 @@ router.beforeEach((to, from, next) => {
     !store.state.loggedIn) {
     // if modal is queried and,
     // the requested modal is only meant to be used post authentication and,
-    // the user is not logged in now, discard the navigation.
-    return next(from)
+    // the user is not logged in now, navigate to home page and set the next query param
+    // to redirect to the requested page after login.
+    return next({ path: '/', query: { next: to.fullPath } })
```

Comment:
The guard logic here makes sure any modals that are part of Post-auth area aren't accessed if user has not logged in yet. (It was a bug-fix added about a couple of years ago)

Since then the app has grown in complexity and loading user-settings data during the initialization takes rather long time now. The cause of the issue was this logic gets executed earlier than user-setting is completely loaded. 
Taking advantage of the existing logic for `next` query param fixes this issue.